pkg/lib/operation/template/local/test/create/operation.go

```diff
@@ -0,0 +1,76 @@
+package create
+
+import (
+	"context"
+	"os"
+
+	"github.com/keboola/go-client/pkg/storageapi"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/template"
+	tmplTest "github.com/keboola/keboola-as-code/internal/pkg/template/test"
+	useTemplate "github.com/keboola/keboola-as-code/pkg/lib/operation/project/local/template/use"
+)
+
+type Options struct {
+	TestName string // name of the test
+	Inputs   template.InputsValues
+}
+
+type dependencies interface {
+	Logger() log.Logger
+	Tracer() trace.Tracer
+}
+
+func Run(ctx context.Context, tmpl *template.Template, o Options, d dependencies) (err error) {
+	tempDir, err := os.MkdirTemp("", "kac-test-template-") //nolint:forbidigo
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := os.RemoveAll(tempDir); err != nil { // nolint: forbidigo
+			d.Logger().Warnf(`cannot remove temp dir "%s": %w`, tempDir, err)
+		}
+	}()
+
+	test, err := tmpl.CreateTest(o.TestName)
+	if err != nil {
+		return err
+	}
+
+	// Save gathered inputs to test file
+	err = test.SaveInputs(o.Inputs.ToMap())
+	if err != nil {
+		return err
+	}
+
+	branchID := 1
+	prjState, _, testDeps, unlockFn, err := tmplTest.PrepareProject(ctx, d.Tracer(), d.Logger(), branchID, false)
+	if err != nil {
+		return err
+	}
+	defer unlockFn()
+	d.Logger().Debugf(`Working directory set up.`)
+
+	// Use template
+	tmplOpts := useTemplate.Options{
+		InstanceName: "test",
+		TargetBranch: model.BranchKey{Id: storageapi.BranchID(branchID)},
+		Inputs:       o.Inputs,
+	}
+	_, _, err = useTemplate.Run(ctx, prjState, tmpl, tmplOpts, testDeps)
+	if err != nil {
+		return err
+	}
+
+	err = test.SaveExpectedOutput(prjState.Fs())
+	if err != nil {
+		return err
+	}
+
+	d.Logger().Infof("The test was created in folder tests/%s.", o.TestName)
+
```

Comment:
Ako na to pozeram v teste, tak `o.Inputs` aj `prjState.Fs()` mozes ziskat pred vytvorenim testu.
Tj. mohli by to byt 2 nove parametre v `CreateTest` a nemoze sa stat, ze by sa to zabudlo vyplnit.

Reply:
@webrouse Takhle? https://github.com/keboola/keboola-as-code/pull/783/commits/c96a760009f0864a5a0342f3e839bb3a75b48ab0 (Testy nám rozbil PS team: https://keboola.slack.com/archives/CQB468K63/p1662722297372559 🙂)

Reply:
Nechcem ta zdrziavat, ale myslel som to trocha inak:
- co potrebujem pre vytvorenie testu? `fs` a `inputs`
- tak si ich pripravim, a vlozim do `CreateTest` (tj. toto volanie by bolo niekde NA KONCI)
- a teda nepotrebujem vobec metody `SaveInputs`, `SaveExpectedOut`, ... a ak ich chcem, aby som to mal podelene, tak mozu byt private
- samostatne metody davaju zmysel vtedy, ak chces tu `err` nejak specialne spracovat, co asi nechces
- alebo ak chces/musis buildovat test postupne, co tiez nechces
- cele moje poznamky boli iba o tom, ako to spravit tak, aby sa to nedalo pouzit nespravne (pri refactoringu, dalsej zmene, ...)
- tu `CreatedTest` strukturu si upravil pekne, ja som si az v teste uvedomil, ze tam nie je vlastne ziaden dovod to buildovat postupne ... ak ti to dava zmysel, tak to uprav, ak nie tak to mozeme nechat takto.

Reply:
Ahaa takto, no, už jsem to tak i chvíli měl, ale potom jsem si říkal že není důvod aplikovat šablonu, když vyfailuje uložení inputů (https://github.com/keboola/keboola-as-code/blob/5a568b7a27b937c3840277efe4bba0b58c7286e0/pkg/lib/operation/template/local/test/create/operation.go#L52-L62). Ale nutný to v zásadě asi není. (Ani nic co by se mělo dít pravidelně.)

Reply:
jj, nemalo by sa to diat casto